### PR TITLE
archive: Fix paramater types

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fix archive module paramater types and remove sanity ignores.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/files/archive.py

### CODE REVIEW

File: plugins/modules/files/archive.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
     description:
       - Remote absolute path, glob, or list of paths or globs for the file or files to compress or archive.
     type: list
+    elements: str
```
Comment (felixfontein) [inline plugins/modules/files/archive.py:26]:
```suggestion
    elements: path
```
Same for the other (and the `elements` in `path`).

Setting both to `path` also allows to remove the `os.path.expanduser` and `os.path.expandvars` calls, and that allows to remove the `use-argspec-type-path` ignore.

Comment (aminvakil) [reply plugins/modules/files/archive.py]: Thanks for the help! Fixed in https://github.com/ansible-collections/community.general/pull/1039/commits/8a31be83468801b7ae8bd59eabdbc98c458e1f3f.

**Comment by aminvakil** [comment]
ready_for_review

**Comment by Andersson007** [comment]
@bendoh could you please take a look?

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
LGTM, so let's merge.

**Comment by felixfontein** [comment]
@aminvakil thanks for fixing this!
@Andersson007 thanks for reviewing!

**Pull request merged**